pages/volunteer/Volunteer.js

```diff
@@ -0,0 +1,122 @@
+    var designSlideQuotes = [];
+    var designSlideNames = [];
+    var designSlideJobTitles = [];
+    VOLUNTEER_INFO.design.forEach(function(designerProfile) {
+      designSlideImages.push(designerProfile.profilePictureImageFilename);
+      designSlideQuotes.push(designerProfile.professionalQuote);
+      designSlideNames.push(designerProfile.fullName);
+      designSlideJobTitles.push(designerProfile.jobTitle);
+    });
+    $scope.designSlideQuotes = designSlideQuotes;
+    $scope.designSlideNames = designSlideNames;
+    $scope.designSlideJobTitles = designSlideJobTitles;
+
+    var developmentSlideImages = [];
+    var developmentSlideQuotes = [];
+    var developmentSlideNames = [];
+    var developmentSlideJobTitles = [];
+    VOLUNTEER_INFO.development.forEach(function(developerProfile) {
+      developmentSlideImages.push(developerProfile.profilePictureImageFilename);
+      developmentSlideQuotes.push(developerProfile.professionalQuote);
+      developmentSlideNames.push(developerProfile.fullName);
+      developmentSlideJobTitles.push(developerProfile.jobTitle);
+    });
+    $scope.developmentSlideQuotes = developmentSlideQuotes;
+    $scope.developmentSlideNames = developmentSlideNames;
+    $scope.developmentSlideJobTitles = developmentSlideJobTitles;
+
+    var researchSlideImages = [];
+    var researchSlideQuotes = [];
+    var researchSlideNames = [];
+    var researchSlideJobTitles = [];
+    VOLUNTEER_INFO.research.forEach(function(researcherProfile) {
+      researchSlideImages.push(researcherProfile.profilePictureImageFilename);
+      researchSlideQuotes.push(researcherProfile.professionalQuote);
+      researchSlideNames.push(researcherProfile.fullName);
+      researchSlideJobTitles.push(researcherProfile.jobTitle);
+    });
+    $scope.researchSlideQuotes = researchSlideQuotes;
+    $scope.researchSlideNames = researchSlideNames;
+    $scope.researchSlideJobTitles = researchSlideJobTitles;
+
+    var marketingSlideImages = [];
+    var marketingSlideQuotes = [];
+    var marketingSlideNames = [];
+    var marketingSlideJobTitles = [];
+    VOLUNTEER_INFO.marketing.forEach(function(marketerProfile) {
+      marketingSlideImages.push(marketerProfile.profilePictureImageFilename);
+      marketingSlideQuotes.push(marketerProfile.professionalQuote);
+      marketingSlideNames.push(marketerProfile.fullName);
+      marketingSlideJobTitles.push(marketerProfile.jobTitle);
+    });
+    $scope.marketingSlideQuotes = marketingSlideQuotes;
+    $scope.marketingSlideNames = marketingSlideNames;
+    $scope.marketingSlideJobTitles = marketingSlideJobTitles;
+
+    var createSlide = function(profileSlides, profileImage, slideId) {
+      profileSlides.push({
+        image: profileImage,
+        id: slideId
+      });
+    };
+    $scope.designSlides = [];
+    $scope.developmentSlides = [];
+    $scope.researchSlides = [];
+    $scope.marketingSlides = [];
+    // Only adding 3 profiles currently.
```

Comment:
Can we randomly select the 3 profiles? I think that would be better.

Reply:
yep! just have to assigned Id to each profile. For now I am just being lazy and using the order in which they are added.

Reply:
Hm, you shouldn't need an ID -- just get the length of the list and do Math.random() or similar (but find a way to avoid duplicates).

Reply:
I mean shoud we not do it hash-table way? Like using ID as the hashed key and just randomly select the key?
Otherwise, we are talking about shuffling an array of dicts here. I don't think thats a good idea.

Reply:
Something along these lines should be fine: https://stackoverflow.com/a/5915122

I.e. do random selection on the _index_. That tells you what element to pull.

Reply:
Ah I was talking getting a random __combination__ of a list. Wouldn't pulling random element like this in a small array result in duplicate and forcing more re-drawing?

Reply:
I was about to do it like this:
1) Get all Ids of a profile into an array
2) Shuffle that array
3) Select first 3 Ids

Does that sound good to you?

Reply:
Ah, I was thinking of select-one-and-remove-from-array, repeat three times. I'm also fine with your approach!

I would also suggest just using the existing indexes, rather than introducing new IDs just for this.

Reply:
Yea that remove operation on array gonna be costly, O(n) for worst case. 

Using hash table would make sense for your approach. But then we are bringing in hash table/map , which is quite memory intensive (imo) , just for randomization purposes.

Reply:
Performance really is not a problem here. We are dealing with <10 elements. So go for whatever is easiest for a dev to understand.

Reply:
With the Async nature of JS, I would have to provide more safe-guard in case of bad randomization no?

Reply:
JS isn't async in that way (for either of the algorithms you mention). As long as you do the steps in sequence (like you might do them in Python), it should work.

Reply:
Done! Can you re-review Volunteer.js before I move onto the carousel directive?